x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java

```diff
@@ -54,6 +66,22 @@ protected AcknowledgedResponse read(StreamInput in) throws IOException {
     @Override
     protected void masterOperation(Task task, DeleteEnrichPolicyAction.Request request, ClusterState state,
                                    ActionListener<AcknowledgedResponse> listener) throws Exception {
+        List<PipelineConfiguration> pipelines = IngestService.getPipelines(state);
+        for (PipelineConfiguration pc: pipelines) {
+            Pipeline pipeline = ingestService.getPipeline(pc.getId());
+
+            for (Processor processor: pipeline.getProcessors()) {
+                if (processor instanceof AbstractEnrichProcessor) {
+                    AbstractEnrichProcessor enrichProcessor = (AbstractEnrichProcessor) processor;
+                    if (enrichProcessor.getPolicyName().equals(request.getName())) {
+                        listener.onFailure(
+                            new ElasticsearchStatusException("Could not delete policy [{}] because a pipeline is referencing it [{}]",
+                                RestStatus.INTERNAL_SERVER_ERROR, request.getName(), pipeline.getId()));
```

Comment:
Should this be BAD_REQUEST instead?

Reply:
I would think `CONFLICT - 409`

Reply:
https://github.com/elastic/elasticsearch/pull/44438/files#diff-a37fe0a3791b5e0c8215af76056efd9eR85